atlascope/core/rest/endpoints/tile_endpoints.py

```diff
@@ -0,0 +1,96 @@
+from django.http.response import HttpResponse
+from django.urls import path
+from drf_yasg import openapi
+from drf_yasg.inspectors import SwaggerAutoSchema
+from drf_yasg.utils import swagger_auto_schema
+from large_image.exceptions import TileSourceError
+from large_image_source_gdal import GDALFileTileSource
+from rest_framework import mixins
+from rest_framework.exceptions import APIException
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from atlascope.core.models import Dataset
+from atlascope.core.rest.additional_serializers import TileMetadataSerializer
+
+
+class TileMetadataView(GenericAPIView, mixins.RetrieveModelMixin):
+    queryset = Dataset.objects.filter(content__isnull=False, dataset_type='tile_source')
+    permission_classes = [IsAuthenticated]
+    serializer_class = TileMetadataSerializer
+
+    def get(self, *args, **kwargs):
+        dataset = self.get_object()
+        tile_source = GDALFileTileSource(f'/vsicurl/use_head=no&url={dataset.content.url}')
+        serializer = self.get_serializer(tile_source)
+        return Response(serializer.data)
+
+
+class TileSchemaGenerator(SwaggerAutoSchema):
+    def get_produces(self):
+        return ['image/png']
```

Comment:
You were right that the `@swagger_auto_schema` decorator provides no way to do this. Even with `extra_overrides`, the `SwaggerAutoSchema` class does not reference its own overrides in its definition of `get_produces`. You've done this the best way it can be done.

Reply:
That's unfortunate. Thanks for double checking my work. I'll add a docstring to this class to explain it's reasoning.

Reply:
Docstring added in 4c790b00d58972e996fb4209f8266a18fc08d72c.